### ISSUE TITLE
Add layer (Clicking on `+`) crashes Avogadro AppImage

### ISSUE THREAD

**Issue by ovari** [OpenChemistry/avogadrolibs#1032]
**Avogadro version:**
 - Avogadrolibs: 1.96.0
 - Qt: 5.15.2
 - AppImage

**Desktop version:**
 - OS: Linux Mint 20.3 Cinnamon

**Describe the bug**

**To Reproduce**
1. Linux Mint operating system in the Hungarian locale
2. Clicking on the `+` (where the mouse pointer is shown in the image below)
3. Avogadro crashes

**Expected behavior**
Clicking on the `+` works.

**Screenshots**
![kép](https://user-images.githubusercontent.com/17465872/179660993-e5c483c1-8e8a-47a3-98a9-f86a95c0791c.png)

Thank you

**Comment by ghutchis** [comment]
Thanks, we'll take a look.

**Comment by ovari** [comment]
Still crashes with https://github.com/OpenChemistry/avogadrolibs/suites/7452934036/artifacts/305076365

**Comment by ovari** [comment]
Still crashes with https://github.com/OpenChemistry/avogadrolibs/suites/7460307004/artifacts/305594276

**Comment by ovari** [comment]
Can this issue please be re-opened?

**Comment by ovari** [comment]
Still crashed with version 1.97.0 from https://nightly.link/OpenChemistry/avogadrolibs/workflows/build_cmake/master/Avogadro2.AppImage.zip

**Comment by ghutchis** [comment]
Weird. I'll look into it .. Is it possible to run the AppImage from a console / terminal to see the output?

**Comment by ovari** [comment]
> Is it possible to run the AppImage from a console / terminal to see the output?

Was unsuccessful with the search engine in finding how to do so.

However, Avogadro2 (Version 1.96.0; Qt Version 5.15.5) flatpak also crashes. Below is the terminal output from Avogadro2 flatpak:
```
$ flatpak run org.openchemistry.Avogadro2 
Gtk-Message: 14:37:07.575: Failed to load module "xapp-gtk3-module"
Qt: Session management error: None of the authentication protocols specified are supported
Locale:  "en_AU"
AvogadroApp Translation  "en_AU"  loaded  "/app/bin/../share/avogadro2/i18n/"
AvogadroLibs Translation  "en_AU"  loaded  "/app/bin/../share/avogadro2/i18n/"
Translated quit:  "&Quit"
Extension plugins dynamically found… 39
Checking for  "commands"  scripts in "/home/c/.var/app/org.openchemistry.Avogadro2/data/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/app/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/usr/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/usr/share/runtime/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/run/host/user-share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/run/host/share/OpenChemistry/Avogadro/commands"
Checking for  "commands"  scripts in "/app/bin/../lib/avogadro2/scripts/commands"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L formats read"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L formats write"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L forcefields"
OBProcess::executeObabel: Running "/app/bin/obabel" "-L charges"
OBProcess::executeObabel: Running "/app/bin/obabel" "-V"
"/app/bin/obabel"  found:  "/app/bin/obabel: Open Babel 3.1.0 -- Jun 23 2022 -- 18:42:58"
Checking for  "inputGenerators"  scripts in "/home/c/.var/app/org.openchemistry.Avogadro2/data/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/app/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/usr/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/usr/share/runtime/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/run/host/user-share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/run/host/share/OpenChemistry/Avogadro/inputGenerators"
Checking for  "inputGenerators"  scripts in "/app/bin/../lib/avogadro2/scripts/inputGenerators"
Checking for  "formatScripts"  scripts in "/home/c/.var/app/org.openchemistry.Avogadro2/data/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/app/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/usr/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/usr/share/runtime/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/run/host/user-share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/run/host/share/OpenChemistry/Avogadro/formatScripts"
Checking for  "formatScripts"  scripts in "/app/bin/../lib/avogadro2/scripts/formatScripts"
Error starting RPC server: "QLocalServer::listen: Address in use"
Open Babel formats ready:  142
Other server is alive. Not starting new instance.
/usr/include/c++/11.3.0/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = Avogadro::Core::LayerData*; _Alloc = std::allocator<Avogadro::Core::LayerData*>; std::vector<_Tp, _Alloc>::reference = Avogadro::Core::LayerData*&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```

The Terminal output after clicking the `+` is:
```
/usr/include/c++/11.3.0/bits/stl_vector.h:1045: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = Avogadro::Core::LayerData*; _Alloc = std::allocator<Avogadro::Core::LayerData*>; std::vector<_Tp, _Alloc>::reference = Avogadro::Core::LayerData*&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```

Hope this helps.

Thank you

**Comment by matterhorn103** [comment]
As far as I can tell this is fixed in 1.99 in both the AppImage and flatpak. At least, I can't reproduce it. Was it fixed and can be closed?

**Comment by ovari** [comment]
Closing as fixed in Avogadro 1.99.0.

![image](https://github.com/user-attachments/assets/4ec2f891-428c-407d-92f7-70e7d7907ea8)

Thank you

**Comment by ghutchis** [comment]
Thanks for all the translation work!
